card/lib/card/query/join.rb

```diff
@@ -28,6 +35,9 @@ def from_and_to opts
             { table: 'cards', alias: object.table_alias }
           when Card::Query::Reference
             { table: 'card_references', alias: object.table_alias }
+          when Card::Query::Join
+            fail "to: cannot be Join" if side == :to
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.